libdirections-hybrid/src/main/java/com/mapbox/navigation/route/hybrid/MapboxHybridRouter.kt

```diff
@@ -1,20 +1,98 @@
 package com.mapbox.navigation.route.hybrid
 
+import android.content.Context
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
 import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.model.Route
 import com.mapbox.navigation.base.route.model.RouteOptionsNavigation
+import com.mapbox.navigation.trip.service.NetworkStatusService
+import com.mapbox.navigation.utils.thread.ThreadController
+import com.mapbox.navigation.utils.thread.monitorChannelWithException
+import java.util.concurrent.atomic.AtomicReference
 
 @MapboxNavigationModule(MapboxNavigationModuleType.HybridRouter, skipConfiguration = true)
 class MapboxHybridRouter(
     private val onboardRouter: Router,
-    private val offboardRouter: Router
+    private val offboardRouter: Router,
+    private val context: Context
 ) : Router {
 
+    private interface RouterDispatchInterface {
+        fun execute(routeOptions: RouteOptionsNavigation, clientCallback: Router.Callback)
+    }
+
+    private class OnBoardRouterHandler(private val offRoute: Router, private val onboaredRouter: Router) : RouterDispatchInterface, Router.Callback {
+        private var offrouterCalled = false
+        private var args: RouteOptionsNavigation? = null
```

Comment:
```suggestion
        private var routeOptions: RouteOptionsNavigation? = null
```
Probably it would be more clear name, because args is too generic, wdyt?

Reply:
👍